### PR TITLE
fix: go mod confliction

### PR DESCRIPTION
**Pull request by Vacant2333**
fix the go mod confliction, and update the readme quick start
FIx: #https://github.com/volcano-sh/volcano-global/issues/4

### CODE REVIEW

**Comment by Vacant2333** [comment]
and fix the readme url
https://github.com/volcano-sh/volcano-global/pull/5

File: installer/dockerfile/controller-manager/Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.21.10 AS builder
+FROM golang:1.22.0 AS builder
```
Comment (hwdef) [inline installer/dockerfile/controller-manager/Dockerfile:15]: It is recommended to use a version with a higher minor version number. Such as 1.22.9

File: installer/dockerfile/scheduler/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.21.10 AS builder
+FROM golang:1.22.0 AS builder
```
Comment (hwdef) [inline installer/dockerfile/scheduler/Dockerfile:15]: same as above

File: installer/dockerfile/webhook-manager/Dockerfile [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.21.10 AS builder
+FROM golang:1.22.0 AS builder
```
Comment (hwdef) [inline installer/dockerfile/webhook-manager/Dockerfile:15]: same as above

File: installer/dockerfile/controller-manager/Dockerfile (hunk 1 above)
Comment (Vacant2333) [reply installer/dockerfile/controller-manager/Dockerfile]: thanks!

File: docs/deploy/README.md [hunk 4]
```diff
@@ -111,7 +111,7 @@ kubectl --context karmada-apiserver apply -f volcano-global-webhooks.yaml
 
 ```bash
 # Switch to Karmada host kubeconfig.
-export KUBECONFIG=/Users/vacant/.kube/karmada.config
+export KUBECONFIG=~/.kube/karmada.config
```
Comment (Monokaix) [inline docs/deploy/README.md:114]: $HOME is better?

File: go.mod [hunk 5]
```diff
@@ -1,25 +1,29 @@
 module volcano.sh/volcano-global
 
-go 1.22.0
+go 1.22.9
+
+toolchain go1.23.0
```
Comment (Monokaix) [inline go.mod:5]: toolchain is no needed.

File: docs/deploy/README.md [hunk 6]
```diff
@@ -67,11 +67,11 @@ we need a plugin to share the target secret with the `volcano-global` namespace.
...
 ```bash
 # Switch to Karmada host kubeconfig.
-export KUBECONFIG=/Users/vacant/.kube/karmada.config
+export KUBECONFIG=$HOME/.kube/karmada.config
 
-# Deploy the Kubernetes Reflector and share the kubeconfig from karmada-system namespace.
+# Deploy the Kubernetes Reflector and share the karmada-webhook-config secret from karmada-system namespace, it includes the kubeconfig of Karmada control plane.
 kubectl --context karmada-host -n kube-system apply -f https://github.com/emberstack/kubernetes-reflector/releases/download/v7.1.262/reflector.yaml
-kubectl --context karmada-host annotate secret kubeconfig \
+kubectl --context karmada-host annotate secret karmada-webhook-config \
```
Comment (Vacant2333) [inline docs/deploy/README.md:74]: Karmada removes the kubeconfig secret, we must get the kubeconfig from the another secret

File: go.mod [hunk 7]
```diff
@@ -1,25 +1,26 @@
...
+	sigs.k8s.io/controller-runtime v0.18.4
 	volcano.sh/apis v1.10.0
 	volcano.sh/volcano v1.10.0
 )
 
 // todo: remove this when resourceBinding suspend merge to master
-// PR ref: https://github.com/karmada-io/karmada/commit/3058bbf50b953d02166625e2b797cb16c7f26396
-replace github.com/karmada-io/karmada => github.com/Vacant2333/karmada v0.0.0-20240606155424-3058bbf50b95
+// Ref: https://github.com/Vacant2333/karmada/tree/suspend-rb-2
+replace github.com/karmada-io/karmada => github.com/Vacant2333/karmada v0.0.0-20241207135009-0a9b0e40a707
```
Comment (Vacant2333) [inline go.mod:23]: change to the correct branch, we will remove this when karmada supports resource binding suspension

**Pull request merged**